### ISSUE TITLE
Can build conda packages for B-Tax release 0.2.5 and 0.2.6

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/B-Tax#225]
@jdebacker,  Thanks for your cooperation in changing the build-related aspects of the B-Tax model.
I was able to use the Package-Builder `pkgbld` package to build and upload to the Anaconda Cloud four (platform-specific) packages for B-Tax release 0.2.5.  Here are the details (remember I don't have any B-Tax source code on my computer):
```
iMac mrh$ conda list pkgbld
# packages in environment at /Users/mrh/anaconda3:
#
# Name                    Version                   Build  Channel
pkgbld                    0.9.0                    py36_0    PSLmodels

iMac mrh$ cat release-btax.py
import pkgbld as pb
pb.release('B-Tax', 'btax', '0.2.5')

iMac mrh$ time python release-btax.py
: Package-Builder building packages for:
:   repository_name = B-Tax
:   package_name = btax
:   model_version = 0.2.5
:   python_versions = ['3.6']
: Package-Builder uploading packages to:
:   Anaconda channel = pslmodels
Cloning into 'B-Tax'...
.
.
.
Using Anaconda API: https://api.anaconda.org
Using "pslmodels" as upload username
Processing 'pkgbld_output/win-64/btax-0.2.5-py36_0.tar.bz2'
Detecting file type...
File type is "conda"
Extracting conda package attributes for upload
Creating package "btax"
Creating release "0.2.5"
Uploading file "pslmodels/btax/0.2.5/win-64/btax-0.2.5-py36_0.tar.bz2"
 uploaded 5156 of 5156Kb: 100.00% ETA: 0.0 minutes
Upload complete
conda package located at:
https://anaconda.org/pslmodels/btax

real	2m13.911s
user	1m6.994s
sys	0m24.153s

iMac mrh$ 
```
Then when I look on Anaconda Cloud, I see this:

<img width="1255" alt="screen shot 2018-11-12 at 1 38 33 pm" src="https://user-images.githubusercontent.com/12170745/48368938-15769f80-e683-11e8-8a32-f105187ecd15.png">

So, now the build-related code in B-Tax is consistent with the other PSL models that are making packages.
Thanks again for your cooperation in all this.

And when I install the new 0.2.5 package from the Anaconda Cloud PSLmodel channel, the `baseline_byasset.csv` generated by `python run_btax_example.py` is exactly the same as the `baseline_byasset_expected.csv` file in the B-Tax repository.

@MattHJensen @rickecon @hdoupe

**Comment by jdebacker** [comment]
@martinholmer I'm glad you were able to successfully build the package.

If I were to try to follow this to build a package at some point in the future, what do I make of the `cat release-btax.py` command?  Is this really `cat>release-btax.py` so that you can type those two lines into the file and save it?

**Comment by martinholmer** [comment]
@jdebacker said in B-Tax issue #225:
> I'm glad you were able to successfully build the packages [for B-Tax release 0.2.5].
>
> If I were to try to follow this to build a package at some point in the future, what [would I do]?

I'll answer that question below.  But first one more thing.

I've made one more change to the PSLmodels Package-Builder tool that makes the handling of release versions much easier than before.  I've changed over Package-Builder, Tax-Calculator, and Behavioral-Response code to be consistent with this new Package-Builder approach to versioning.  You can see the changes I made to Behavioral-Responses code in [B-R pull request 21](https://github.com/open-source-economics/Behavioral-Responses/pull/21).  Can you make those kinds of changes to the B-Tax code?  After removing the `B-Tax/versioneer.py` and `B-Tax/btax/_version.py` files, you need to eliminate any reference to them and change the `setup.py` and `btax/__init__.py` code to be like in the Behavioral-Responses repo.  Do a recursive grep to find (and eliminate) any other stray references to `versioneer` or `_version` in B-Tax code.  Let me know if you have any questions about making these changes.

While you're working on those changes, I'm going to be adding user documentation for the Package-Builder tool.  Don't try this just yet, but here's the approach that I've been using on my computer:
```
$ conda install -c PSLmodels  pkgbld --yes
...

$ pbrelease --version
Package-Builder 0.12.0

$ pbrelease B-Tax btax 0.2.6
...
```

Once we've got B-Tax compliant with PSL package-building criteria, then we can turn to making OG-USA compliant.

@rickecon @MattHJensen

**Comment by jdebacker** [comment]
@martinholmer I can do these things this week, but I'll be at a conference so my workflow will be intermittent.

**Comment by martinholmer** [comment]
@jdebacker said in B-Tax issue #225:
> The changes you suggested have been merged with PR #226 and a new release has been created, 0.2.6. Please let me know if you have any issues building a package.

The package-building went smoothly. And using the new `btax` 0.2.6 package produces the same expected results from the `run_btax_example.py` script as before.  What your changes have gained is clearer versions:
instead of something like this (when using versioneer in 0.2.5):
```
>>> import btax
>>> btax.__version__
'0.2.5+0.g0c7409b.dirty'
```
we now get this:
```
>>> import btax
>>> btax.__version__
'0.2.6'
```
and will always get this even if many B-Tax commits have been made since version 0.2.6 was released.
